gammapy/analysis/core.py

```diff
@@ -252,6 +252,20 @@ def _map_making(self):
         maker_safe_mask = SafeMaskMaker(
             methods=safe_mask_selection, **safe_mask_settings
         )
+
+        bkg_maker_config = {}
+        if self.config.datasets.background.exclusion:
+            exclusion_region = Map.read(self.config.datasets.background.exclusion)
+            bkg_maker_config["exclusion_mask"] = exclusion_region
+        bkg_maker_config.update(self.config.datasets.background.parameters)
+
+        bkg_method = self.config.datasets.background.method
+        if bkg_method == "fov_background":
+            log.debug(f"Creating FoVBackgroundMaker with arguments {bkg_maker_config}")
+            bkg_maker = FoVBackgroundMaker(**bkg_maker_config)
+        else:
+            bkg_maker = None
+
         stacked = MapDataset.create(geom=geom, name="stacked", **geom_irf)
```

Comment:
Does it make sense to fetch the exclusion_region when no background method is provided?
I would suggest to re-use the same coding pattern found in `_spectrum_extraction` function below.

```
        bkg_method = self.config.datasets.background.method
        if bkg_method == "fov_background":
            bkg_maker_config = {}
            if datasets_settings.background.exclusion:
                exclusion_region = Map.read(datasets_settings.background.exclusion)
                bkg_maker_config["exclusion_mask"] = exclusion_region
            
            bkg_maker_config.update(self.config.datasets.background.parameters)
            bkg_maker = FoVBackgroundMaker(**bkg_maker_config)
            log.debug(
                f"Creating FoVBackgroundMaker with arguments {bkg_maker_config}"
            )
        else:
            bkg_maker = None
            log.warning(
                f"No background maker set for 3d analysis. Check configuration."
            )
```